pkgs/development/tools/continuous-integration/gitlab-runner/default.nix

```diff
@@ -17,18 +17,19 @@ buildGoModule rec {
   # For patchShebangs
   buildInputs = [ bash ];
 
-  vendorHash = "sha256-sP9lboy1M4+AB1Z0VsX5eNxZ9ckuJBgEPxavAbOSk8g=";
+  vendorHash = "sha256-Rzy4R4QR+rPqzhjZlqcuiP3DDLOu9Z2fb42WPaSPR/4=";
 
   src = fetchFromGitLab {
     owner = "gitlab-org";
     repo = "gitlab-runner";
     rev = "v${version}";
-    sha256 = "sha256-gTjGf/WuMoGNvsiWWj3Y7l2MvSzBM12RY/Odolm4fNw=";
+    sha256 = "sha256-GMvBZ3H29F4XyisAt3J4VWRwaEIF7ZQ/tI0gKbDTS/E=";
   };
 
   patches = [
     ./fix-shell-path.patch
     ./remove-bash-test.patch
+    ./fix-invalid-host-header.patch # see https://gitlab.com/gitlab-org/gitlab-runner/-/merge_requests/4249
```

Comment:
It's preferred to use `fetchpatch` instead of commitint the patch to nixpkgs.

I think you should be able to use https://gitlab.com/gitlab-org/gitlab-runner/-/merge_requests/4249.patch as the url.

See https://github.com/NixOS/nixpkgs/blob/6039decbd1c40e31b25a71fefe23b007e9fc92d4/pkgs/development/libraries/librest/1.0.nix#L30 for an example.

Reply:
@michaeladler explained that the patch had to be edited/backported to apply cleanly on top of 16.2.1. There's nothing that could be fetchpatch'd here.

Reply:
Ok got it.